packages/utils/src/carequality/cq-monthly-stats/shared.ts

```diff
@@ -0,0 +1,140 @@
+import * as dotenv from "dotenv";
+dotenv.config();
+// keep that ^ on top
+import { getEnvVarOrFail } from "@metriport/core/util/env-var";
+import { initReadonlyDbPool } from "@metriport/core/util/sequelize";
+import { QueryTypes } from "sequelize";
+
+const sqlDBCreds = getEnvVarOrFail("DB_CREDS");
+const sqlReadReplicaEndpoint = getEnvVarOrFail("DB_READ_REPLICA_ENDPOINT");
+
+export const readOnlyDBPool = initReadonlyDbPool(sqlDBCreds, sqlReadReplicaEndpoint);
+
+export type RequestParams = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  cqDirectory: any[];
+  endOfPreviousMonth: string;
+  dayIndex: number;
+};
+
+export type GWStats = {
+  nonErroredResponses?: string;
+  totalDocRetrieved?: string;
+  avgResponseTimeMs?: number;
+};
+
+export type GWWithStats = {
+  [gw: string]: GWStats;
+};
+
+export type ImplementerStats = {
+  implementer?: string;
+  gw?: string;
+  avgResponseTimeMs?: number;
+  nonErroredResponses?: string;
+  totalDocRetrieved?: string;
+};
+
+export type CountPerGW = { [key: string]: [number] };
+
+export async function queryResultsTable<TableResults>(
+  tableName: string,
+  endOfPreviousMonth: string,
+  dayIndex: number
+): Promise<TableResults[]> {
+  const dayPlusOne = dayIndex + 1;
+
+  console.log(`Querying for day ${dayPlusOne}...`);
+
+  const query = `
+      SELECT * FROM ${tableName}
```

Comment:
Use bind parameters to avoid SQL injections

Example: 
Declare tableName const. Add a bind parameter:
```
SELECT * FROM :tableName
WHERE....
```
Then, add replacements to the query:
```
await readOnlyDBPool.query(query, {
    type: QueryTypes.SELECT,
    replacements: tableName
  });
```

^ might not be exactly like that, but something along these lines. And same applies for all params in the query